### ISSUE TITLE
Move away from coffeescript in favor of ES 2015

### ISSUE THREAD

**Issue by cmckni3** [immense/knockout-datatable#37]

**Comment by cmckni3** [comment]
@cmbankester Is this finished?

**Comment by cmbankester** [comment]
No, unfortunately, it is not. It's been on the backburner for effectively a year now. Take a look at the [feature/refactor](https://github.com/immense/knockout-datatable/compare/feature/refactor) branch. I did make a lot of progress, but even some of that progress is now dated (e.g. gulp).

Dammit, JavaScript ecosystem.

Mainly what needs to be done at this point is testing the converted code.
